frontend/app/src/App.test.tsx

```diff
@@ -2448,5 +2510,23 @@ describe("App", () => {
         message: "random string",
       })
     })
+
+    it("properly handles TERMINATE_WEBSOCKET_CONNECTION and RESTART_WEBSOCKET_CONNECTION messages", () => {
+      prepareHostCommunicationManager()
+
+      const originalConnectionManager = getMockConnectionManager()
+
+      fireWindowPostMessage({
+        type: "TERMINATE_WEBSOCKET_CONNECTION",
+      })
+
+      expect(originalConnectionManager.disconnect).toHaveBeenCalled()
+
+      fireWindowPostMessage({
+        type: "RESTART_WEBSOCKET_CONNECTION",
+      })
+
+      expect(getMockConnectionManager()).not.toBe(originalConnectionManager)
```

Comment:
I couldn't get testing that `newConnectionManager.connect` was called to work for whatever reason. I think it has something to do with the call to `.connect` happening in the constructor, which probably isn't running normally due to the class being mocked.

This likely won't be a big deal because we have e2e test coverage for this message type to ensure this happens.